server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -0,0 +1,271 @@
+
+    /**
+     * Creates a new HealthPeriodicLogger.
+     * This creates a scheduled job using the SchedulerEngine framework and runs it on the current health node.
+     *
+     * @param settings the cluster settings, used to get the interval setting.
+     * @param clusterService the cluster service, used to know when the health node changes.
+     * @param client the client used to call the Health Service.
+     * @param healthService the Health Service, where the actual Health API logic lives.
+     */
+    public HealthPeriodicLogger(Settings settings, ClusterService clusterService, Client client, HealthService healthService) {
+        this.settings = settings;
+        this.clusterService = clusterService;
+        this.client = client;
+        this.healthService = healthService;
+        this.clock = Clock.systemUTC();
+        this.pollInterval = HEALTH_PERIODIC_LOGGER_POLL_INTERVAL_SETTING.get(settings);
+    }
+
+    /**
+     * Initializer method to avoid the publication of a self reference in the constructor.
+     */
+    public void init() {
+        clusterService.addListener(this);
+        clusterService.getClusterSettings()
+            .addSettingsUpdateConsumer(HEALTH_PERIODIC_LOGGER_POLL_INTERVAL_SETTING, this::updatePollInterval);
+    }
+
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        // wait for the cluster state to be recovered
+        if (event.state().blocks().hasGlobalBlock(GatewayService.STATE_NOT_RECOVERED_BLOCK)) {
+            return;
+        }
+
+        DiscoveryNode healthNode = HealthNode.findHealthNode(event.state());
+        if (healthNode == null) {
+            this.isHealthNode = false;
+            this.maybeCancelJob();
+            return;
+        }
+        final boolean isCurrentlyHealthNode = healthNode.getId().equals(this.clusterService.localNode().getId());
+        if (this.isHealthNode != isCurrentlyHealthNode) {
+            this.isHealthNode = isCurrentlyHealthNode;
+            if (this.isHealthNode) {
+                // we weren't the health node, and now we are
+                maybeScheduleJob();
+            } else {
+                // we were the health node, and now we aren't
+                maybeCancelJob();
+            }
+        }
+    }
+
+    @Override
+    public void close() {
+        SchedulerEngine engine = scheduler.get();
+        if (engine != null) {
+            engine.stop();
+        }
+    }
+
+    @Override
+    public void triggered(SchedulerEngine.Event event) {
+        if (event.getJobName().equals(HEALTH_PERIODIC_LOGGER_JOB_NAME)) {
+            this.tryToLogHealth();
+        }
+    }
+
+    // default visibility for testing purposes
+    void tryToLogHealth() {
+        if (this.currentlyRunning.compareAndExchange(false, true) == false) {
+            // We ensure that the flag will only be released once when it's time
+            RunOnce release = new RunOnce(() -> currentlyRunning.set(false));
```

Comment:
I think this comment can be removed. I added because this was a suggestion in the PR and it wouldn't be easy to click around to see what this does.